Istio update to use proxy image with path normalization

jrajahalme opened the pull request:
Update Istio GSG and CI tests to use Cilium versions of Istio proxy (1.1.3) and pilot (1.1.2).

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/7712)
<!-- Reviewable:end -->

ianvernon commented on Documentation/gettingstarted/istio.rst:
```diff
@@ -99,6 +97,18 @@ Deploy Istio onto Kubernetes:
+
+::
+
+    $ kubectl get crds | grep 'istio.io\|certmanager.k8s.io' | wc -l
```
Would be nice to have something like `watch -n 1 "kubectl get crds | grep 'istio.io\|certmanager.k8s.io' | wc -l"` instead so users don't have to keep entering the same command again and again.

jrajahalme replied:
I'll try that and update accordingly!

jrajahalme committed:
istio: Update istio proxy to 1.1.3

Use proxy with path normalization (1.1.3)

Istio release 1.1.3 is not out yet, but we can use the 1.1.3 proxy
with Istio 1.1.2 to get the path normalization fix in.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

jrajahalme committed:
test: Update Istio test to 1.1.2 with proxy 1.1.3.

Replace the istio-cilium.yaml file with a new one, generated with helm
using the updated Istio GSG.

Replace the bookinfo manifests with ones newly generated with helm.

Istio release 1.1.3 is not out yet, but we can use the 1.1.3 proxy
with Istio 1.1.2 to get the path normalization fix in.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

nebril commented on Documentation/gettingstarted/istio.rst:
```diff
@@ -71,21 +81,20 @@ Cilium's Envoy filters to query the Cilium agent for policy configuration:
           --set global.controlPlaneSecurityEnabled=true \
           --set global.mtls.enabled=true \
-          --set global.proxy.image=proxy_debug \
+          --set global.proxy.image=docker.io/cilium/istio_proxy:1.1.3 \
```
is this supposed to be `1.1.3` and not `1.1.2`?

jrajahalme replied:
1.1.3 is correct here.

nebril commented on test/k8sT/istio.go:
```diff
@@ -90,13 +93,22 @@ var _ = Describe("K8sIstioTest", func() {
+		res.ExpectSuccess("unable to create Istio CRDs")
+
+		By("Waiting for Istio CRDs to be ready")
+		err := kubectl.WaitForCRDCount("istio.io|certmanager.k8s.io", 53, helpers.HelperTimeout)
```
53 looks really magical here, can we move it out to a variable with descriptive name?

jrajahalme replied:
Its the number of Istio CRDs created by version 1.1.2. I'll add a variable if any other updates are needed, thanks!

nebril reviewed (commented):
Two nitpicks left in comments, but not really critical, please fix if you think it's worth it.

ianvernon commented on examples/kubernetes-istio/istio-sidecar-injector-configmap-debug.yaml:
```diff
@@ -9,7 +9,7 @@ data:
       initContainers:
       - name: istio-init
-        image: docker.io/cilium/istio_proxy_init:0.7.0
+        image: docker.io/cilium/istio_proxy_init:1.1.2
```
Why not 1.1.3?
